Kanakpandit17/search bar

kanakpandit17 opened the pull request:
## What does this PR do?
The search bar is made properly visible on the top right corner of the webpage. Also, it has placeholder so that users can understand the purpose of the search bar. The focus-ring is fully visible on clicking.

Solved issue #8693

Type of change: UI Improvement

kanakpandit17 committed:
Added change

kanakpandit17 committed:
Added placeholder, increased the size of searchbar, and made the focus ring fully visible on getting clicked.

CarinaWolli commented:
@Jaibles do we want to make changed to the search bar? That's how the changes of this PR look: 
<img width="1237" alt="Screenshot 2023-05-05 at 15 53 11" src="https://user-images.githubusercontent.com/30310907/236492910-aa0ead35-89e4-444d-be83-2c7960b487e9.png">

sean-brydon reviewed (approved):
Tested and happy with this :) Thanks @kanakpandit17

PeerRich commented:
![CleanShot 2023-05-05 at 22 33 50@2x](https://user-images.githubusercontent.com/8019099/236572528-330e6888-d8ae-4b4d-965d-07d51606c3df.jpg)
the margin-right looks odd

PeerRich reviewed (changes requested):
![CleanShot 2023-05-05 at 22 33 50@2x](https://user-images.githubusercontent.com/8019099/236572606-ea0ffab4-1a39-45b2-ab2a-2cdbfacddeaa.jpg)
the margin right looks odd

PeerRich commented on apps/web/pages/apps/index.tsx:
```diff
@@ -37,12 +37,13 @@ function AppsSearch({
+      containerClassName={classNames("focus:!ring-offset-0 pr-4 pl-6", className)}
       type="search"
       autoComplete="false"
+      placeholder="Search App Name"
```
we need i18n strings here

PeerRich commented on apps/web/pages/apps/index.tsx:
```diff
@@ -37,12 +37,13 @@ function AppsSearch({
   return (
     <TextField
-      className="bg-subtle !border-muted !pl-0 focus:!ring-offset-0"
+    className="bg-subtle !border-muted !pl-15 pr-6 focus:!ring-offset-0 md:w-96"
```
there is a prettier issue here

kanakpandit17 commented:
will make the required changes

kanakpandit17 committed:
Made search bar more elegant than before

CarinaWolli commented:
@kanakpandit17 we should make the search bar wide enough to show the whole text, right now the text is cut off:
<img width="269" alt="Screenshot 2023-05-08 at 15 13 30" src="https://user-images.githubusercontent.com/30310907/236833522-869a6b81-1460-489e-8081-a299a871ba33.png">

Jaibles commented:
> @Jaibles do we want to make changed to the search bar? That's how the changes of this PR look: <img alt="Screenshot 2023-05-05 at 15 53 11" width="1237" src="https://user-images.githubusercontent.com/30310907/236492910-aa0ead35-89e4-444d-be83-2c7960b487e9.png">

This above is far too big imo. We can't solve every visibility problem by making everything in the app bigger. @kanakpandit17 I'd be happy with any width between `190px` (current width in live) and `220px` (width in Figma) but not bigger. Given that most searches are short and results are instant there's enough space to type and you're likely to see what you need in a few key strokes.

If we want placeholder we can just use "Search..."
<img width="1145" alt="CleanShot 2023-05-08 at 14 41 03@2x" src="https://user-images.githubusercontent.com/4536123/236839873-15110050-ca20-482f-aa55-367788510314.png">
.

kanakpandit17 commented:
> This above is far too big IMO and the text is too far away from the magnify icon (it should only be an 8px gap). @kanakpandit17 I'd be happy with any width between `190px` (current width in live) and `220px` ([width in Figma](https://www.figma.com/file/xk4HOxtSI82J0F7enMxeak/Cal---Live?node-id=26%3A53096&t=dygkprGcVLgpfZib-1)) but not bigger. Given that most searches are short and results are instant there's enough space to type and you're likely to see what you need in a few key strokes. We can't solve every visibility problem by making everything in the app bigger. The search already has a prominent position and we have an app-wide search in the top of the sidebar and accessible through k-bar.

Actually text is completely visible in my search bar, will check out

kanakpandit17 committed:
Update index.tsx

Added required placeholder. Also adjusted width to 200px. Also made the gap of 8px  between search icon and placeholder.

Jaibles commented:
@kanakpandit17 Any if this issue on mobile is fixed in your pr?

https://user-images.githubusercontent.com/4536123/236951037-8789b61f-cbe6-4780-b09d-5bce9ab89193.MOV

Udit-takkar reviewed (commented):
@kanakpandit17 this issue has been fixed by some other contributor. you can check at latest main branch

<img width="1440" alt="Screenshot 2023-05-10 at 2 44 53 PM" src="https://github.com/calcom/cal.com/assets/53316345/9e92cdae-aa9e-4bb6-8035-8eb8819661ce">
<img width="814" alt="Screenshot 2023-05-10 at 2 45 16 PM" src="https://github.com/calcom/cal.com/assets/53316345/d6e72af6-de21-4ead-8729-5227c88b99c7">

CarinaWolli commented:
As @Udit-takkar mentioned, a fix for this already been merged to main (https://github.com/calcom/cal.com/pull/8741)
